app/src/main/java/com/zeapo/pwdstore/PasswordStore.kt

```diff
@@ -658,81 +711,35 @@ class PasswordStore : AppCompatActivity(R.layout.activity_pwdstore) {
-                                    R.string.password_exists_message,
-                                    destinationLongName,
-                                    sourceLongName)
-                                )
-                                .setPositiveButton(R.string.dialog_ok) { _, _ ->
-                                    movePasswords(source, destinationFile, sourceLongName, destinationLongName)
-                                }
-                                .setNegativeButton(R.string.dialog_cancel, null)
-                                .show()
-                        } else {
-                            movePasswords(source, destinationFile, sourceLongName, destinationLongName)
-                        }
-                    }
-                    resetPasswordList()
-                    if (plist != null) {
-                        plist!!.dismissActionMode()
-                    }
-                }
             }
         }
         super.onActivityResult(requestCode, resultCode, data)
     }
 
-    private fun movePasswords(source: File, destinationFile: File, sourceLongName: String, destinationLongName: String) {
+    private suspend fun movePasswords(source: File, destinationFile: File, sourceLongName: String, destinationLongName: String) {
         val sourceDestinationMap = if (source.isDirectory) {
             // Recursively list all files (not directories) below `source`, then
             // obtain the corresponding target file by resolving the relative path
             // starting at the destination folder.
-            val sourceFiles = source.listFilesRecursively()
-            sourceFiles.associateWith { destinationFile.resolve(it.relativeTo(source)) }
+            source.listFilesRecursively().associateWith { destinationFile.resolve(it.relativeTo(source)) }
         } else {
             mapOf(source to destinationFile)
         }
         if (!source.renameTo(destinationFile)) {
-            // TODO this should show a warning to the user
             e { "Something went wrong while moving." }
+            withContext(Dispatchers.Main) {
+                MaterialAlertDialogBuilder(this@PasswordStore)
+                    .setTitle(R.string.password_move_error_title)
+                    .setMessage(getString(R.string.password_move_error_message, source, destinationFile))
+                    .setCancelable(true)
+                    .setPositiveButton(android.R.string.ok, null)
+                    .show()
+            }
         } else {
             AutofillMatcher.updateMatches(this, sourceDestinationMap)
-            commitChange(resources
-                .getString(
-                    R.string.git_commit_move_text,
-                    sourceLongName,
-                    destinationLongName))
+            withContext(Dispatchers.Main) {
+                commitChange(resources.getString(R.string.git_commit_move_text, sourceLongName, destinationLongName))
```

Comment:
Could this be changed to commit everything a single time after the move is done?

Reply:
Yeah I think it can be done, how do you propose I structure the commit message like?

Reply:
I think we could keep it as is for single file moves and simply make it something like `Rename/Move multiple passwords/files` for multiple files. For complex moves, I believe `git show` to be a better tool than just `git log`, we don't need to duplicate the information.

Reply:
Hm yeah that makes sense. Will fix it after work.